src/db/PromoMonth/promoMonth.ts

```diff
@@ -0,0 +1,30 @@
+import { DbOperationError } from "../../errorClasses";
+import { getCollection } from "../../db/mongoCollections";
+import type { Place as GooglePlace } from "@googlemaps/google-maps-services-js";
+import { PromotionMonth } from "../../db/types";
+
+export type GetPromoMonthArgs = {
+	placeID: GooglePlace["place_id"];
+	month: number;
+	year: number;
+};
+
+// find a promo by place id and month and year
+export async function getPromoMonth({
+	placeID,
+	month,
+	year,
+}: GetPromoMonthArgs): Promise<PromotionMonth> {
+	if (!(placeID && month && year)) throw new DbOperationError("Invalid input");
+	const { _col, _connection } = await getCollection<PromotionMonth>("promotionMonth");
+
+	const promoMonth = await _col.findOneAndUpdate(
+		{ placeID, month, year },
+		{ $setOnInsert: { placeID, month, year, totalSpent: 0 } },
+		{ upsert: true, returnDocument: "after" }
+	);
+
+	await _connection.close();
+
+	return promoMonth.value!;
```

Comment:
@carpenterd777 yeah it's because normally this can be null, but since i have `upsert: true` then it actually can't be null. I guess maybe their types don't account for upsert. I'll add a comment explaining it.